File: 2350-find-closest-number-to-zero/find-closest-number-to-zero.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findClosestNumber = function(nums) {
    let num = Infinity
    let dis = Infinity
    for(let i of nums){
        let d = Math.abs(i)
        if(d<=dis){
            dis = d
            if(num==dis){
                num = Math.max(num,i)
            }else{
                num = i
            }
        }
    }
    return num
};

File: 2350-find-closest-number-to-zero/find-closest-number-to-zero.py
class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        num = float("inf")
        dis = float("inf")
        for i in nums:
            d = abs(i)
            if d <= dis:
                dis = d
                if dis == num:
                    num = max(num,i)
                else:
                    num = i
        return num